src/sgame/sg_buildable.cpp

```diff
@@ -181,6 +181,25 @@ bool G_InsideBase(gentity_t *self) {
 	return trap_InPVSIgnorePortals(self->s.origin, mainBuilding->s.origin);
 }
 
+bool G_DretchCanDamageEntity( const gentity_t *self, const gentity_t *ent )
```

Comment:
why not start by an assert that checks that ent has correct type (pre-conditions) and then do a single "return player || ( building && attackable )"?

I think it would be easier to read (and use less lines)?

Also, why using a pointer for ent and self, instead of references?

Reply:
about the pointers, basically because the code around it uses pointer already, so it's more consistent.

I've reworked the `if(cond) return true; else return false;`, but kept the switch. I still find it more readable that way.